tests/components/suez_water/conftest.py

```diff
@@ -13,3 +42,14 @@ def mock_setup_entry() -> Generator[AsyncMock]:
         "homeassistant.components.suez_water.async_setup_entry", return_value=True
     ) as mock_setup_entry:
         yield mock_setup_entry
+
+
+@pytest.fixture
+def mock_coordinator(hass: HomeAssistant) -> SuezWaterCoordinator:
+    """Create mock coordinator."""
+
+    return SuezWaterCoordinator(
+        hass,
+        None,
+        MOCK_DATA["counter_id"],
+    )
```

Comment:
We should not mock the coordinator like this. I'll explain why in a few comments